### ISSUE TITLE
Get different result using pycocotools or code without json (Pose mAP@0.5~0.95: 0.552 vs. 0.405)

### ISSUE THREAD

**Issue by shupinghu** [ultralytics/ultralytics#4436]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When I validate the offical model "yolov8s-pose.pt", I find the printed results is totally different (Pose mAP@0.5~0.95: 0.552 vs. 0.405) as:
![yECcIkxCn9](https://github.com/ultralytics/ultralytics/assets/52944151/4c4ccd75-2077-4c51-97a7-05c8b39edaba)
![VbKSrf7pA6](https://github.com/ultralytics/ultralytics/assets/52944151/753858ce-2e24-45d4-adb4-31e5a516ded1)

I would like to ask that why the difference occurs?  When I don't have a json file, how can I get the correct result?

### Additional

This is my validation code:

from ultralytics import YOLO                                            
model = YOLO("yolov8s-pose.yaml")  # build a new model from scratch
model = YOLO("yolov8s-pose.yaml").load("./yolov8s-pose.pt")  # load a pretrained model (recommended for training)
metrics = model.val(data="coco-pose.yaml", imgsz=320, conf=0.5, iou=0.5)  # evaluate model performance on the validation set-------line 5

The validation datasets:
coco/val2017/, json file was downloaded in offical coco website, and txt labels was downloaded in https://github.com/ultralytics/yolov5/releases/download/v1.0/coco2017labels-pose.zip, 

Envs:
torch == 1.8.0
torchvision == 0.9.0
pycocotools == 2.0.7
numpy == 1.24.4
......

other exp:
exp2: Change line 5 to "metrics = model.val(data="coco-pose.yaml", imgsz=640)" , I can get a close results on pose map, but the bbox results are still totally different:
![pgZhulefav](https://github.com/ultralytics/ultralytics/assets/52944151/4bead27f-cd3d-42fe-9af8-9313f410a654)
![ApYIWV4YUL](https://github.com/ultralytics/ultralytics/assets/52944151/dc2aa49d-3499-41ab-9193-cc448b1b328a)

exp3: Change line 5 to "metrics = model.val(data="coco-pose.yaml", imgsz=640, conf=0.5)",  I can even get a higher pose map than in exp2 (default conf = 0.001) using code without json, and the bbox map is almost consistent, it is very strange:
![ppDxZRfgIP](https://github.com/ultralytics/ultralytics/assets/52944151/09598094-917d-42b6-92a7-a1c59575c2c4)
![NUdkBCsyKa](https://github.com/ultralytics/ultralytics/assets/52944151/fc99ee7e-a3dd-4855-97f9-a8ca0d1ef055)
